### PR TITLE
Use BPM memory limits for container resource limits

### PR DESCRIPTION
**Pull request by kramerul**
It turned out that doppler uses a lot of memory in a setup with 2 replicas. 
Therefore, we would like be able to limit the memory usage. Using the existing BPM configuration variables for container limits would help.

### CODE REVIEW

File: pkg/bosh/converter/container_factory.go [hunk 1]
```diff
@@ -515,6 +518,15 @@ func bpmProcessContainer(
...
 	command, args := generateBPMCommand(&process, postStart)
+	limits := corev1.ResourceList{}
+	if process.Limits.Memory != "" {
+		quantity, err := resource.ParseQuantity(process.Limits.Memory)
+		if err != nil {
+			log.Errorf(context.TODO(), "Error parsing %s: %v", process.Limits.Memory, err)
+		} else {
+			limits[corev1.ResourceMemory] = quantity
+		}
+	}
```
Comment (loewenstein) [inline pkg/bosh/converter/container_factory.go]: Just wondering, why the `limits` don't come from where the `requests` come from.

Comment (kramerul) [reply pkg/bosh/converter/container_factory.go]: The `Limits` structure in the `BPMProcess` was available before our change for the `Requests`. Therefore, I decided to use the data provided by `BPMProcess.Limits`

**Comment by viovanov** [comment]
@kramerul lgtm - can you please add a test?

**Comment by kramerul** [comment]
@viovanov , I added a negative test. A test was already included in the existing BPM resource test.

**Pull request merged**